tools/serverpod_cli/lib/src/generator/dart/library_generators/model_library_generator.dart

```diff
@@ -373,41 +416,44 @@ class SerializableModelLibraryGenerator {
   ) {
     return Method(
       (m) {
-        m
-          ..name = 'copyWith'
-          ..annotations.add(refer('override'))
-          ..optionalParameters.addAll(
-            fields.where((field) => field.shouldIncludeField(serverCode)).map(
-              (field) {
-                var fieldType = field.type.reference(
-                  serverCode,
-                  nullable: true,
-                  subDirParts: classDefinition.subDirParts,
-                  config: config,
-                );
-
-                var type = field.type.nullable ? refer('Object?') : fieldType;
-                var defaultValue =
-                    field.type.nullable ? const Code('_Undefined') : null;
-
-                return Parameter((p) {
-                  p
-                    ..name = field.name
-                    ..named = true
-                    ..type = type
-                    ..defaultTo = defaultValue;
-                });
-              },
-            ),
-          )
-          ..returns = refer(classDefinition.className)
-          ..body = refer(classDefinition.className)
-              .call(
-                [],
-                _buildCopyWithAssignment(classDefinition, fields),
-              )
-              .returned
-              .statement;
+        m.name = 'copyWith';
+        if (!classDefinition.isParentClass) {
+          m.annotations.add(refer('override'));
+        }
+        m.optionalParameters.addAll(
+          [...classDefinition.parentFields, ...fields]
```

Comment:
Question: Also another nit thing here, since you are destructing the classDefinition.parentFields and merge it with fields, can we simply not do this at the top level instead and just have the input fields be the merged fields? 

What is the difference between parentFields and fields?

Reply:
I think we could, we only need to know which fields are from parent and which fields are from the child class because we have to pass them to `super` or `this` respectively

Reply:
Got it, then my suggestion here is to lift it up wherever it is possible.

Reply:
Resolved in Slack, this will be cleaned up when the table related classes will be activated